servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Executors.java

```diff
@@ -254,4 +256,23 @@ public static Executor from(java.util.concurrent.Executor jdkExecutor,
         return EXECUTOR_PLUGINS.wrapExecutor(
                 new DefaultExecutor(jdkExecutor, scheduledExecutorService, mayInterruptOnCancel));
     }
+
+    /**
+     * Wraps the given {@link Executor} with the provided {@link TimeSource}.
+     * <p>
+     * The given {@link Executor} is wrapped into a {@link DelegatingExecutor} which delegates all calls but
+     * {@link Executor#currentTime(TimeUnit)} which will go to the provided {@link TimeSource}.
+     *
+     * @param delegate the {@link Executor} which should use the provided time source.
+     * @param timeSource the {@link TimeSource} that should be used when calling {@link Executor#currentTime(TimeUnit)}.
+     * @return The wrapped {@link Executor} with a custom {@link TimeSource}.
+     */
+    public static Executor withTimeSource(final Executor delegate, final TimeSource timeSource) {
+        return new DelegatingExecutor(delegate) {
+            @Override
+            public long currentTime(final TimeUnit unit) {
+                return timeSource.currentTime(unit);
```

Comment:
It's not enough just to swap the `TimeSource` because all scheduled tasks will still depend on the real time.
Since the primary use-case is testing, consider not adding this on the main public API. 
There is already a test-fixture `TestExecutor` that allows control of the time source and execution of the tasks.